### PR TITLE
Fix #2853: require at least one parent after `extends`

### LINKED ISSUE #2853: Parsing: `extends` with 0 parents compiles
Following bug is discovered by @nicolasstucki during a discussion.

The compiler should produce an error message, but it accepts the code below:

```Scala
class A extends { }
object A extends { }
```

### PR DESCRIPTION
**Pull request by liufengyun**
Fix #2853: require at least one parent after `extend`

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/parsing/Parsers.scala [hunk 1]
```diff
@@ -2312,7 +2312,12 @@ object Parsers {
     /** TemplateOpt = [`extends' Template | TemplateBody]
      */
     def templateOpt(constr: DefDef): Template =
-      if (in.token == EXTENDS) { in.nextToken(); template(constr)._1 }
+      if (in.token == EXTENDS) {
+        val offset = in.skipToken()
+        val (tmpl, _) = template(constr)
+        if (tmpl.parents.isEmpty) syntaxError("parents expected after `extend`", offset)
```
Comment (nicolasstucki) [inline compiler/src/dotty/tools/dotc/parsing/Parsers.scala]: `extend` -> `extends`

**Comment by nicolasstucki** [comment]
You caught a bug in test file `tests/run/t6090.scala`

**Comment by nicolasstucki** [comment]
LGTM

**Comment by DarkDimius** [comment]
We've decided to keep this behavior:
 - it's consistent with the grammar
 - it's consistent with scala2 behavior.

**Pull request closed without merging**